### ISSUE TITLE
Bang-string properties keys are not always caught

### ISSUE THREAD

**Issue by teutoburg** [AstarVienna/astar-utils#4]
The method `NestedMapping.update()` has a clause with a comment that reads:
```python
# Catch any bang-string properties keys
```

From this I'd assume an input dict like e.g.

```python
yamldict = {
    'alias': 'OBS',
    'properties': {
        'temperature': 100,
        '!SIM.someglobal': True,
        },
    }
```

would put `'!SIM.someglobal'` into the top level as a new sub-dict. However,

```python
nestmap = NestedMapping(yamldict)
```

outputs:

```python
NestedMapping({'OBS': {'temperature': 100, '!SIM.someglobal': True}})
```

whereas we would expect the global key to end up in the top level, such that the output becomes:

```python
NestedMapping({'OBS': {'temperature': 100}, 'SIM': {'someglobal': True}})
```

The reason for this is that `NestedMapping.update()` only calls `recursive_update()` for "properties" input dicts, so the code never ends up in the clause mentioned in the beginning.

We need to figure out which way around `ScopeSim` expects this to work. I see to possibilities:

1. The comment is correct and such bang-keys from `"properties"` _should_ end up in the top level, but don't. In this case that'll be a relatively easy fix, but we should maybe consider all such cases, not just from within the "magic" `"properties"` key.
2. The code is working correctly, and bang-keys from `"properties"` _should not_ end up in the top level. In that case it might be best to just remove the comment and the corresponding clause.

Full minimum example:

```python
from astar_utils.nested_mapping import NestedMapping

yamldict = {
    'alias': 'OBS',
    'properties': {
        'temperature': 100,
        '!SIM.someglobal': True,
        },
    }

nestmap = NestedMapping(yamldict)
print(f"{nestmap!r}")
```

BTW, _updating_ an existing instance works as expected:

```python

from astar_utils.nested_mapping import NestedMapping

yamldict = {
    'alias': 'OBS',
    'properties': {
        'temperature': 100,
        },
    }

nestmap = NestedMapping(yamldict)

new_yamldict = {
    '!OBS.temperature': 42,
    '!SIM.someglobal': True,
    }

nestmap.update(new_yamldict)
print(f"{nestmap!r}")
```

**Comment by hugobuddel** [comment]
I don't think this makes sense:
```yaml
yamldict = {
    'alias': 'OBS',
    'properties': {
        'temperature': 100,
        '!SIM.someglobal': True,
        },
    }
```
Because then the `OBS` dictionary sets values in the `SIM` dictionary. It sounds to me like that should not be allowed. Or are there cases where something like this already happens?

It should be fine to use bang-strings as value though, maybe catching those is what is intended? (Even though the comment says 'keys'.)

**Comment by teutoburg** [comment]
After a quick look at this with @astronomyk on Friday, we discovered the following:

There are three YAMLs in the METIS IRDB package that use the infamous `DetectorModePropertiesSetter` effect, and in those, there are indeed keys which intentionally contain `!XYZ.abc`-style strings. As the effect name suggests, the intention here was to set keys somewhere else. However, the `DetectorModePropertiesSetter` effect was already disliked before (for some other reason that escapes me now) and the consensus between @astronomyk and myself was to try to come up with something more elegant at some point, but not now.

Since this functionality of using bang-strings in _keys_ to set something somewhere else is confusing at best, and only used in that one very specific case, we concluded it might be best to leave things as they are for now, but include a warning in that part of the code, so we know if it accidentally occurs somewhere else (which I don't expect). Once we find a better solution for what METIS needs, we can update the package and then the warning from ScopeSim will basically mean "update your METIS package".

Hope all of this makes some sense, it's a bit convoluted...
